aiida/tools/dbimporters/plugins/materialsproject.py

```diff
@@ -0,0 +1,198 @@
+# -*- coding: utf-8 -*-
+###########################################################################
+# Copyright (c), The AiiDA team. All rights reserved.                     #
+# This file is part of the AiiDA code.                                    #
+#                                                                         #
+# The code is hosted on GitHub at https://github.com/aiidateam/aiida_core #
+# For further information on the license, see the LICENSE.txt file        #
+# For further information please visit http://www.aiida.net               #
+###########################################################################
+"""Module that contains the class definitions necessary to offer support for
+queries to Materials Project."""
+
+from __future__ import absolute_import
+import os
+import datetime
+import requests
+from pymatgen import MPRester
+from aiida.tools.dbimporters.baseclasses import CifEntry, DbEntry, DbImporter, DbSearchResults
+
+
+class MaterialsProjectImporter(DbImporter):
+    """
+    Database importer for the Materials Project.
+    """
+
+    _properties = 'structure'
+    _supported_keywords = None
+
+    def __init__(self, api_key=None):
+        """
+        Instantiate the MaterialsProjectImporter by setting up the Materials API (MAPI) connection details
+
+        :param api_key: the API key to be used to access MAPI
+        """
+        self.setup_db(api_key=api_key)
```

Comment:
This might just pass all the ``kwargs``.

Reply:
Yep, we might as well do that at this stage. Fixed.